git-repository/examples/init-non-bare-repo.rs

```diff
@@ -0,0 +1,33 @@
+// cargo run -p git-repository --example new
+
+use anyhow::Context;
+use git_repository as git;
+
+fn main() -> anyhow::Result<()> {
+    let work_dir = std::env::args_os()
```

Comment:
curious... why `args_os` and not `args`?  does that help with platform-specific path handling?

Reply:
Yes, paths aren't necessarily UTF-8 encoded and thus can't be forced into a `String`. `gitoxide` does a lot to deal with paths correctly and not paint itself into a corner by assuming encodings that aren't there, to match the way git does it as a bare minimum and be just as flexible.

Funnily enough `git` will decode windows paths as UTF-8 and last time I checked just assumes that always works probably causing strange behavior. `gitoxide` will panic in these cases.